### ISSUE TITLE
[Github] complaints of deprecation notice for authentication via URL query parameters

### ISSUE THREAD

**Issue by tonythomas01** [pennersr/django-allauth#2532]
Message from Github on using the current system: 
```
On May 7th, 2020 at 09:23 (UTC) your application (***) used an access token (with the User-Agent python-requests/2.23.0) as part of a query parameter to access an endpoint through the GitHub API:

https://api.github.com/user

Please use the Authorization HTTP header instead as using the `access_token` query parameter is deprecated.

Depending on your API usage, we'll be sending you this email reminder on a monthly basis.

Visit https://developer.github.com/changes/2020-02-10-deprecating-auth-through-query-param for more information about suggested workarounds and removal dates.
``` 
allauth-version: django-allauth==0.41.0

I see its fixed on master though.

**Comment by heliodor** [comment]
See #2458

Available in 0.42.0

**Comment by pennersr** [comment]
https://codeberg.org/allauth/django-allauth/issues/2532
